tests/integration/test_upgrade.py

```diff
@@ -78,20 +79,48 @@ async def test_upgrade(kubernetes_cluster: juju.model.Model, ops_test: OpsTest):
     }
     bundle, _ = await Bundle.create(ops_test)
     charms = await bundle.discover_charm_files(ops_test)
-    k8s: juju.application.Application = kubernetes_cluster.applications["k8s"]
+    k8s: juju.application.Application = kubernetes_cluster.applications[CONTROL_PLANE_APP]
 
     @retry(
         stop=stop_after_attempt(10),
         wait=wait_fixed(30),
         before_sleep=before_sleep_log(log, logging.WARNING),
     )
-    async def _wait_for_idle():
+    async def _wait_for_kube_system_pods():
         """Wait for the model to become idle."""
         kube_system_pods = await get_rsc(k8s.units[0], "pods", namespace="kube-system")
         assert all(
             p["status"]["phase"] == "Running" for p in kube_system_pods
         ), "Kube-system not yet ready"
 
+    @retry(
+        stop=stop_after_attempt(10),
+        wait=wait_fixed(30),
+        before_sleep=before_sleep_log(log, logging.WARNING),
+    )
+    async def _wait_for_upgrade_complete():
+        """Wait for the model to become idle."""
+        k8s_apps = {
+            k: v
+            for k, v in kubernetes_cluster.applications.items()
+            if v.startswith(CONTROL_PLANE_APP)
```

Comment:
Shouldn't this be:
```suggestion
            if k.startswith(CONTROL_PLANE_APP)
```
IIRC, `applications` returns `app_name, Application`